src/NLog/LogFactory.cs

```diff
@@ -351,6 +351,24 @@ public Logger GetCurrentClassLogger()
         }
 
         /// <summary>
+        /// Gets the logger with the name of the current class. 
+        /// </summary>
+        /// <returns>The logger.</returns>
+        /// <remarks>This is a slow-running method. 
+        /// Make sure you're not doing this in a loop.</remarks>
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public T GetCurrentClassLogger<T>() where T : Logger
```

Comment:
CLS compliant is not needed for generic types, but it is for this class (as it's not marked als CLScompliant false). 

Thanks for checking! 

PS. The return type is here `T` ;)